### ISSUE TITLE
.NET process filtering is too restrictive

### ISSUE THREAD

**Issue by tmds** [Azure/vscode-kubernetes-tools#1003]
Currently, any process that is not named `dotnet` is filtered out:

https://github.com/Azure/vscode-kubernetes-tools/blob/268260093d0bd3265d981fb571261d3c4818c436/src/debug/dotNetDebugProvider.ts#L81-L84

Since .NET 3.1 (released 2 years back), .NET applications come with a native executable that launches the app. These apphost are not named `dotnet`.

```
$ dotnet new web -o web
$ cd web
$ dotnet build
$ bin/Debug/net6.0/web 
info: Microsoft.Hosting.Lifetime[14]
      Now listening on: http://localhost:5000
info: Microsoft.Hosting.Lifetime[14]
      Now listening on: https://localhost:5001
info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Production
info: Microsoft.Hosting.Lifetime[0]
      Content root path: /tmp/web
```

A way of filtering .NET apps is to look at the `/tmp/dotnet-diagnostic-<pid>-*-socket` files.
```
$ pidof web
16873
$ ls /tmp/dotnet-diagnostic-$(pidof web)-*-socket
/tmp/dotnet-diagnostic-16873-1204654-socket
```

Alternatively, the filtering could be left out, since there are not that many processes expected to be running in a container anyhow.

**Comment by tmds** [comment]
cc @itowlson

**Comment by itowlson** [comment]
Hmm, I'm not too familiar with the .NET debugging component.  @tedchamb you were the original author of this feature - are you able to take a look please?

**Comment by itowlson** [comment]
@pantosha Sorry to pester you, but the original author of this code isn't responding right now, and I know you have some knowledge of this area.  Would you be able to take a look at this issue too?  Thanks ever so much!

**Comment by pantosha** [comment]
@itowlson,  I will take a look

**Comment by tmds** [comment]
> A way of filtering .NET apps is to look at the /tmp/dotnet-diagnostic-<pid>-*-socket files.

@pantosha this is a good way to filter .NET apps. Though it has some complexity to it. Alternatives are to do no filtering, or another alternative would be to allow specifying the process name with a setting (which defaults to `dotnet`).

**Comment by tmds** [comment]
@pantosha are you still looking into the issue?

@itowlson the simplest fix may be to allow specifying the process name, and default to `dotnet` to match the current behavior.

**Comment by pantosha** [comment]
@tmds yeah. I reviewed current implementation. I didn't find a way to use socket files without deeply rewriting not only .net debugger, but also debuggers for other languages.

I will prepare PR that removes restrictions within a week.
